### DISCUSSION TITLE
[Bug] Git `fetch.pruneTags = true` makes scoop unable to update

### DISCUSSION THREAD

**Discussion by m-lamonaca** [ScoopInstaller/Scoop#5701] [Help and Questions] [answered]
#### Current Behavior

With the git config `fetch.pruneTags = true` scoop fails to update with only the error "Update failed".

#### Expected Behavior
Scoop updates without problems.

#### Additional context/output
With `fetch.pruneTags = false` (which is the also git default) the problem does not happen.

### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:** PowerShell 7.3.7

#### Scoop Configuration

```json
{
  "last_update": "2023-10-09T09:34:49.1296383+02:00",
  "scoop_branch": "master",
  "scoop_repo": "https://github.com/lukesampson/scoop"
}
```

**Comment by rashil2000** [comment]
I'm unable to repro this: 

```console
~                                        in [pwsh] at 67% 19:10:19
❯ git config --global push.followTags                   7GiB/15GiB
true

~                                        in [pwsh] at 67% 19:10:31
❯ scoop update                                          8GiB/15GiB
Updating Scoop...
Updating Buckets...
Scoop was updated successfully!

```

**Reply by m-lamonaca** [reply]
Sorry, I'm an idiot and realized now that to config is `fetch.pruneTags` not `followTags`. I will update the question for clarity

**Reply by rashil2000** [reply]
Oh, I see. What is your use case for using the `fetch.pruneTags = true` setting?

**Reply by m-lamonaca** [reply]
Simply keep the tags tidy and up to date with the remote by automatically pruning them, nothing more specific.

It's not a deal breaker if it's not supported. 
I reported it mainly because, to me, it appeared strange that a seemingly unrelated git config changes Scoop's behavior.

**Reply by rashil2000** [reply]
Do you require that setting for all repos? Since it sounds useful, it might be helpful to enable it on a per-repo basis, and not globally (so as to not affect all repos).

**Reply by m-lamonaca** [reply]
Having it enabled system-wide makes it easier to use but it's not required

**Comment by HUMORCE** [comment, marked as answer]
```
cd (scoop prefix scoop)
git config --local fetch.pruneTags false
gci "$(scoop prefix scoop)\..\..\..\buckets" -dir | foreach { git -C "$_" config --local fetch.pruneTags false }
```

If specific git config items are affecting Scoop's ability to function properly, Scoop should set them for the repo of Core/Buckets when updating, because Scoop is deeply bound to Git.

**Reply by m-lamonaca** [reply]
Thanks! This worked perfectly
